2016-12-18 19:58:19,770 - INFO  - [main:BaseTestCase@51] - Starting ZooKeeper Standalone Server: 11221
2016-12-18 19:58:19,777 - INFO  - [main:BaseTestCase@54] - ZooKeeper server started
2016-12-18 19:58:19,808 - INFO  - [Thread-0:Environment@100] - Client environment:zookeeper.version=3.4.8--1, built on 02/06/2016 03:18 GMT
2016-12-18 19:58:19,809 - INFO  - [Thread-0:Environment@100] - Client environment:host.name=Lenovo-Robin
2016-12-18 19:58:19,809 - INFO  - [Thread-0:Environment@100] - Client environment:java.version=1.8.0_111
2016-12-18 19:58:19,809 - INFO  - [Thread-0:Environment@100] - Client environment:java.vendor=Oracle Corporation
2016-12-18 19:58:19,809 - INFO  - [Thread-0:Environment@100] - Client environment:java.home=/usr/lib/jvm/java-8-oracle/jre
2016-12-18 19:58:19,809 - INFO  - [Thread-0:Environment@100] - Client environment:java.class.path=/home/robin/Travail/SDTD/DistributedLogProcessing/zookeeper-master-worker/target/test-classes:/home/robin/Travail/SDTD/DistributedLogProcessing/zookeeper-master-worker/target/classes:/home/robin/.m2/repository/org/slf4j/slf4j-api/1.7.1/slf4j-api-1.7.1.jar:/home/robin/.m2/repository/junit/junit/4.11/junit-4.11.jar:/home/robin/.m2/repository/org/hamcrest/hamcrest-core/1.3/hamcrest-core-1.3.jar:/home/robin/.m2/repository/org/apache/zookeeper/zookeeper/3.4.8/zookeeper-3.4.8.jar:/home/robin/.m2/repository/org/slf4j/slf4j-log4j12/1.6.1/slf4j-log4j12-1.6.1.jar:/home/robin/.m2/repository/log4j/log4j/1.2.16/log4j-1.2.16.jar:/home/robin/.m2/repository/jline/jline/0.9.94/jline-0.9.94.jar:/home/robin/.m2/repository/io/netty/netty/3.7.0.Final/netty-3.7.0.Final.jar:/home/robin/.m2/repository/org/apache/curator/curator-recipes/2.1.0-incubating/curator-recipes-2.1.0-incubating.jar:/home/robin/.m2/repository/org/apache/curator/curator-framework/2.1.0-incubating/curator-framework-2.1.0-incubating.jar:/home/robin/.m2/repository/org/apache/curator/curator-client/2.1.0-incubating/curator-client-2.1.0-incubating.jar:/home/robin/.m2/repository/com/google/guava/guava/14.0.1/guava-14.0.1.jar:
2016-12-18 19:58:19,810 - INFO  - [Thread-0:Environment@100] - Client environment:java.library.path=/usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib
2016-12-18 19:58:19,810 - INFO  - [Thread-0:Environment@100] - Client environment:java.io.tmpdir=/tmp
2016-12-18 19:58:19,810 - INFO  - [Thread-0:Environment@100] - Client environment:java.compiler=<NA>
2016-12-18 19:58:19,810 - INFO  - [Thread-0:Environment@100] - Client environment:os.name=Linux
2016-12-18 19:58:19,810 - INFO  - [Thread-0:Environment@100] - Client environment:os.arch=amd64
2016-12-18 19:58:19,811 - INFO  - [Thread-0:Environment@100] - Client environment:os.version=4.4.0-53-generic
2016-12-18 19:58:19,811 - INFO  - [Thread-0:Environment@100] - Client environment:user.name=robin
2016-12-18 19:58:19,811 - INFO  - [Thread-0:Environment@100] - Client environment:user.home=/home/robin
2016-12-18 19:58:19,813 - INFO  - [Thread-0:Environment@100] - Client environment:user.dir=/home/robin/Travail/SDTD/DistributedLogProcessing/zookeeper-master-worker
2016-12-18 19:58:19,824 - INFO  - [Thread-0:ZooKeeper@438] - Initiating client connection, connectString=localhost:11221 sessionTimeout=10000 watcher=org.apache.zookeeper.book.TestAssignmentRecovery$5@26c89ad1
2016-12-18 19:58:19,851 - INFO  - [Thread-0-SendThread(localhost:11221):ClientCnxn$SendThread@1032] - Opening socket connection to server localhost/127.0.0.1:11221. Will not attempt to authenticate using SASL (unknown error)
2016-12-18 19:58:19,896 - INFO  - [Standalone server with clientPort:11221:QuorumPeerConfig@103] - Reading configuration from: /tmp/test765414132113960671.junit.dir/zoo.cfg
2016-12-18 19:58:19,897 - INFO  - [Standalone server with clientPort:11221:ZooKeeperServerMain@95] - Starting server
2016-12-18 19:58:19,904 - INFO  - [Standalone server with clientPort:11221:Environment@100] - Server environment:zookeeper.version=3.4.8--1, built on 02/06/2016 03:18 GMT
2016-12-18 19:58:19,904 - INFO  - [Standalone server with clientPort:11221:Environment@100] - Server environment:host.name=Lenovo-Robin
2016-12-18 19:58:19,904 - INFO  - [Standalone server with clientPort:11221:Environment@100] - Server environment:java.version=1.8.0_111
2016-12-18 19:58:19,904 - INFO  - [Standalone server with clientPort:11221:Environment@100] - Server environment:java.vendor=Oracle Corporation
2016-12-18 19:58:19,904 - INFO  - [Standalone server with clientPort:11221:Environment@100] - Server environment:java.home=/usr/lib/jvm/java-8-oracle/jre
2016-12-18 19:58:19,904 - INFO  - [Standalone server with clientPort:11221:Environment@100] - Server environment:java.class.path=/home/robin/Travail/SDTD/DistributedLogProcessing/zookeeper-master-worker/target/test-classes:/home/robin/Travail/SDTD/DistributedLogProcessing/zookeeper-master-worker/target/classes:/home/robin/.m2/repository/org/slf4j/slf4j-api/1.7.1/slf4j-api-1.7.1.jar:/home/robin/.m2/repository/junit/junit/4.11/junit-4.11.jar:/home/robin/.m2/repository/org/hamcrest/hamcrest-core/1.3/hamcrest-core-1.3.jar:/home/robin/.m2/repository/org/apache/zookeeper/zookeeper/3.4.8/zookeeper-3.4.8.jar:/home/robin/.m2/repository/org/slf4j/slf4j-log4j12/1.6.1/slf4j-log4j12-1.6.1.jar:/home/robin/.m2/repository/log4j/log4j/1.2.16/log4j-1.2.16.jar:/home/robin/.m2/repository/jline/jline/0.9.94/jline-0.9.94.jar:/home/robin/.m2/repository/io/netty/netty/3.7.0.Final/netty-3.7.0.Final.jar:/home/robin/.m2/repository/org/apache/curator/curator-recipes/2.1.0-incubating/curator-recipes-2.1.0-incubating.jar:/home/robin/.m2/repository/org/apache/curator/curator-framework/2.1.0-incubating/curator-framework-2.1.0-incubating.jar:/home/robin/.m2/repository/org/apache/curator/curator-client/2.1.0-incubating/curator-client-2.1.0-incubating.jar:/home/robin/.m2/repository/com/google/guava/guava/14.0.1/guava-14.0.1.jar:
2016-12-18 19:58:19,905 - INFO  - [Standalone server with clientPort:11221:Environment@100] - Server environment:java.library.path=/usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib
2016-12-18 19:58:19,905 - INFO  - [Standalone server with clientPort:11221:Environment@100] - Server environment:java.io.tmpdir=/tmp
2016-12-18 19:58:19,905 - INFO  - [Standalone server with clientPort:11221:Environment@100] - Server environment:java.compiler=<NA>
2016-12-18 19:58:19,905 - INFO  - [Standalone server with clientPort:11221:Environment@100] - Server environment:os.name=Linux
2016-12-18 19:58:19,906 - INFO  - [Standalone server with clientPort:11221:Environment@100] - Server environment:os.arch=amd64
2016-12-18 19:58:19,906 - INFO  - [Standalone server with clientPort:11221:Environment@100] - Server environment:os.version=4.4.0-53-generic
2016-12-18 19:58:19,906 - INFO  - [Standalone server with clientPort:11221:Environment@100] - Server environment:user.name=robin
2016-12-18 19:58:19,906 - INFO  - [Standalone server with clientPort:11221:Environment@100] - Server environment:user.home=/home/robin
2016-12-18 19:58:19,906 - INFO  - [Standalone server with clientPort:11221:Environment@100] - Server environment:user.dir=/home/robin/Travail/SDTD/DistributedLogProcessing/zookeeper-master-worker
2016-12-18 19:58:19,924 - INFO  - [Standalone server with clientPort:11221:ZooKeeperServer@787] - tickTime set to 2000
2016-12-18 19:58:19,925 - INFO  - [Standalone server with clientPort:11221:ZooKeeperServer@796] - minSessionTimeout set to -1
2016-12-18 19:58:19,925 - INFO  - [Standalone server with clientPort:11221:ZooKeeperServer@805] - maxSessionTimeout set to -1
2016-12-18 19:58:19,933 - INFO  - [Standalone server with clientPort:11221:NIOServerCnxnFactory@89] - binding to port 0.0.0.0/0.0.0.0:11221
2016-12-18 19:58:19,944 - INFO  - [Thread-0-SendThread(localhost:11221):ClientCnxn$SendThread@876] - Socket connection established to localhost/127.0.0.1:11221, initiating session
2016-12-18 19:58:19,944 - INFO  - [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:11221:NIOServerCnxnFactory@192] - Accepted socket connection from /127.0.0.1:51366
2016-12-18 19:58:19,960 - INFO  - [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:11221:ZooKeeperServer@900] - Client attempting to establish new session at /127.0.0.1:51366
2016-12-18 19:58:19,960 - WARN  - [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:11221:NIOServerCnxnFactory@213] - Ignoring unexpected runtime exception
java.lang.NullPointerException
	at org.apache.zookeeper.server.ZooKeeperServer.createSession(ZooKeeperServer.java:569)
	at org.apache.zookeeper.server.ZooKeeperServer.processConnectRequest(ZooKeeperServer.java:902)
	at org.apache.zookeeper.server.NIOServerCnxn.readConnectRequest(NIOServerCnxn.java:418)
	at org.apache.zookeeper.server.NIOServerCnxn.readPayload(NIOServerCnxn.java:198)
	at org.apache.zookeeper.server.NIOServerCnxn.doIO(NIOServerCnxn.java:244)
	at org.apache.zookeeper.server.NIOServerCnxnFactory.run(NIOServerCnxnFactory.java:203)
	at java.lang.Thread.run(Thread.java:745)
2016-12-18 19:58:20,970 - INFO  - [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:11221:ZooKeeperServer@900] - Client attempting to establish new session at /127.0.0.1:51366
2016-12-18 19:58:20,977 - INFO  - [SyncThread:0:FileTxnLog@199] - Creating new log file: log.1
2016-12-18 19:58:21,033 - INFO  - [SyncThread:0:ZooKeeperServer@645] - Established session 0x159134c88bd0000 with negotiated timeout 10000 for client /127.0.0.1:51366
2016-12-18 19:58:21,034 - INFO  - [Thread-0-SendThread(localhost:11221):ClientCnxn$SendThread@1299] - Session establishment complete on server localhost/127.0.0.1:11221, sessionid = 0x159134c88bd0000, negotiated timeout = 10000
2016-12-18 19:58:21,037 - INFO  - [Thread-0-EventThread:TestAssignmentRecovery$5@196] - Event: WatchedEvent state:SyncConnected type:None path:null
2016-12-18 19:58:21,214 - INFO  - [Thread-0-EventThread:RecoveredAssignments$1@108] - Getting tasks for recovery
2016-12-18 19:58:21,217 - INFO  - [Thread-0-EventThread:RecoveredAssignments$3@155] - Getting worker assignments for recovery: 1
2016-12-18 19:58:21,219 - INFO  - [Thread-0-EventThread:RecoveredAssignments$4@238] - Getting statuses for recovery
2016-12-18 19:58:21,253 - INFO  - [Thread-0-EventThread:RecoveredAssignments$7@359] - Task correctly deleted: /assign/worker-002/task-002
2016-12-18 19:58:21,261 - INFO  - [Thread-0-EventThread:RecoveredAssignments$7@359] - Task correctly deleted: /assign/worker-002
2016-12-18 19:58:21,261 - INFO  - [Thread-0-EventThread:RecoveredAssignments$8@384] - Processing assignments for recovery
2016-12-18 19:58:21,262 - INFO  - [Thread-0-EventThread:RecoveredAssignments@397] - Size of tasks: 3
2016-12-18 19:58:21,262 - INFO  - [Thread-0-EventThread:RecoveredAssignments@400] - Assignment: task-001
2016-12-18 19:58:21,262 - INFO  - [Thread-0-EventThread:RecoveredAssignments@405] - Size of tasks after assignment filtering: 2
2016-12-18 19:58:21,262 - INFO  - [Thread-0-EventThread:RecoveredAssignments@412] - Size of tasks after status filtering: 2
2016-12-18 19:58:21,262 - INFO  - [Thread-0-EventThread:TestAssignmentRecovery$6@228] - Completed recovery: 0
2016-12-18 19:58:21,270 - INFO  - [Thread-0-EventThread:RecoveredAssignments$7@359] - Task correctly deleted: /tasks/task-001
2016-12-18 19:58:21,312 - INFO  - [main:NIOServerCnxn@1008] - Closed socket connection for client /127.0.0.1:51366 which had sessionid 0x159134c88bd0000
2016-12-18 19:58:21,312 - INFO  - [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:11221:NIOServerCnxnFactory@219] - NIOServerCnxn factory exited run method
2016-12-18 19:58:21,313 - INFO  - [Standalone server with clientPort:11221:ZooKeeperServer@469] - shutting down
2016-12-18 19:58:21,313 - INFO  - [Thread-0-SendThread(localhost:11221):ClientCnxn$SendThread@1158] - Unable to read additional data from server sessionid 0x159134c88bd0000, likely server has closed socket, closing socket connection and attempting reconnect
2016-12-18 19:58:21,313 - INFO  - [Standalone server with clientPort:11221:SessionTrackerImpl@225] - Shutting down
2016-12-18 19:58:21,313 - INFO  - [Standalone server with clientPort:11221:PrepRequestProcessor@767] - Shutting down
2016-12-18 19:58:21,314 - INFO  - [Standalone server with clientPort:11221:SyncRequestProcessor@209] - Shutting down
2016-12-18 19:58:21,314 - INFO  - [ProcessThread(sid:0 cport:11221)::PrepRequestProcessor@143] - PrepRequestProcessor exited loop!
2016-12-18 19:58:21,314 - INFO  - [SyncThread:0:SyncRequestProcessor@187] - SyncRequestProcessor exited!
2016-12-18 19:58:21,314 - INFO  - [Standalone server with clientPort:11221:FinalRequestProcessor@415] - shutdown of request processor complete
2016-12-18 19:58:21,318 - INFO  - [main:BaseTestCase@51] - Starting ZooKeeper Standalone Server: 11222
2016-12-18 19:58:21,319 - INFO  - [main:BaseTestCase@54] - ZooKeeper server started
2016-12-18 19:58:21,320 - WARN  - [Standalone server with clientPort:11222:ZooKeeperServerMain@76] - Unable to register log4j JMX control
javax.management.InstanceAlreadyExistsException: log4j:hiearchy=default
	at com.sun.jmx.mbeanserver.Repository.addMBean(Repository.java:437)
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerWithRepository(DefaultMBeanServerInterceptor.java:1898)
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerDynamicMBean(DefaultMBeanServerInterceptor.java:966)
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerObject(DefaultMBeanServerInterceptor.java:900)
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerMBean(DefaultMBeanServerInterceptor.java:324)
	at com.sun.jmx.mbeanserver.JmxMBeanServer.registerMBean(JmxMBeanServer.java:522)
	at org.apache.zookeeper.jmx.ManagedUtil.registerLog4jMBeans(ManagedUtil.java:53)
	at org.apache.zookeeper.server.ZooKeeperServerMain.initializeAndRun(ZooKeeperServerMain.java:74)
	at org.apache.zookeeper.book.StandaloneServer$TestZKSMain.initializeAndRun(StandaloneServer.java:51)
	at org.apache.zookeeper.book.StandaloneServer.run(StandaloneServer.java:97)
2016-12-18 19:58:21,320 - INFO  - [Thread-2:ZooKeeper@438] - Initiating client connection, connectString=localhost:11222 sessionTimeout=10000 watcher=org.apache.zookeeper.book.TestAssignmentRecovery$3@e412f21
2016-12-18 19:58:21,321 - INFO  - [Standalone server with clientPort:11222:QuorumPeerConfig@103] - Reading configuration from: /tmp/test8706253066308715605.junit.dir/zoo.cfg
2016-12-18 19:58:21,322 - INFO  - [Standalone server with clientPort:11222:ZooKeeperServerMain@95] - Starting server
2016-12-18 19:58:21,322 - INFO  - [Standalone server with clientPort:11222:ZooKeeperServer@787] - tickTime set to 2000
2016-12-18 19:58:21,322 - INFO  - [Standalone server with clientPort:11222:ZooKeeperServer@796] - minSessionTimeout set to -1
2016-12-18 19:58:21,322 - INFO  - [Standalone server with clientPort:11222:ZooKeeperServer@805] - maxSessionTimeout set to -1
2016-12-18 19:58:21,323 - INFO  - [Standalone server with clientPort:11222:NIOServerCnxnFactory@89] - binding to port 0.0.0.0/0.0.0.0:11222
2016-12-18 19:58:21,323 - INFO  - [Thread-2-SendThread(localhost:11222):ClientCnxn$SendThread@1032] - Opening socket connection to server localhost/127.0.0.1:11222. Will not attempt to authenticate using SASL (unknown error)
2016-12-18 19:58:21,324 - INFO  - [Thread-2-SendThread(localhost:11222):ClientCnxn$SendThread@876] - Socket connection established to localhost/127.0.0.1:11222, initiating session
2016-12-18 19:58:21,325 - INFO  - [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:11222:NIOServerCnxnFactory@192] - Accepted socket connection from /127.0.0.1:43568
2016-12-18 19:58:21,325 - INFO  - [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:11222:ZooKeeperServer@900] - Client attempting to establish new session at /127.0.0.1:43568
2016-12-18 19:58:21,326 - INFO  - [SyncThread:0:FileTxnLog@199] - Creating new log file: log.1
2016-12-18 19:58:21,371 - INFO  - [SyncThread:0:ZooKeeperServer@645] - Established session 0x159134c8e0d0000 with negotiated timeout 10000 for client /127.0.0.1:43568
2016-12-18 19:58:21,372 - INFO  - [Thread-2-SendThread(localhost:11222):ClientCnxn$SendThread@1299] - Session establishment complete on server localhost/127.0.0.1:11222, sessionid = 0x159134c8e0d0000, negotiated timeout = 10000
2016-12-18 19:58:21,372 - INFO  - [Thread-2-EventThread:TestAssignmentRecovery$3@133] - Event: WatchedEvent state:SyncConnected type:None path:null
2016-12-18 19:58:21,414 - INFO  - [Thread-0-EventThread:TestAssignmentRecovery$5@196] - Event: WatchedEvent state:Disconnected type:None path:null
2016-12-18 19:58:21,566 - INFO  - [Thread-2-EventThread:RecoveredAssignments$1@108] - Getting tasks for recovery
2016-12-18 19:58:21,569 - INFO  - [Thread-2-EventThread:RecoveredAssignments$3@155] - Getting worker assignments for recovery: 1
2016-12-18 19:58:21,571 - INFO  - [Thread-2-EventThread:RecoveredAssignments$4@238] - Getting statuses for recovery
2016-12-18 19:58:21,575 - INFO  - [ProcessThread(sid:0 cport:11222)::PrepRequestProcessor@651] - Got user-level KeeperException when processing sessionid:0x159134c8e0d0000 type:delete cxid:0x16 zxid:0x12 txntype:-1 reqpath:n/a Error Path:/assign/worker-002 Error:KeeperErrorCode = Directory not empty for /assign/worker-002
2016-12-18 19:58:21,581 - INFO  - [Thread-2-EventThread:RecoveredAssignments$7@359] - Task correctly deleted: /assign/worker-002/task-003
2016-12-18 19:58:21,582 - ERROR - [Thread-2-EventThread:RecoveredAssignments$7@362] - Failed to delete task dataorg.apache.zookeeper.KeeperException$NotEmptyException: KeeperErrorCode = Directory not empty for /assign/worker-002
2016-12-18 19:58:21,582 - INFO  - [Thread-2-EventThread:RecoveredAssignments$7@359] - Task correctly deleted: /assign/worker-002/task-002
2016-12-18 19:58:21,582 - INFO  - [Thread-2-EventThread:RecoveredAssignments$7@359] - Task correctly deleted: /assign/worker-002
2016-12-18 19:58:21,583 - INFO  - [Thread-2-EventThread:RecoveredAssignments$8@384] - Processing assignments for recovery
2016-12-18 19:58:21,583 - INFO  - [Thread-2-EventThread:RecoveredAssignments@397] - Size of tasks: 3
2016-12-18 19:58:21,583 - INFO  - [Thread-2-EventThread:RecoveredAssignments@400] - Assignment: task-001
2016-12-18 19:58:21,583 - INFO  - [Thread-2-EventThread:RecoveredAssignments@405] - Size of tasks after assignment filtering: 2
2016-12-18 19:58:21,586 - INFO  - [Thread-2-EventThread:RecoveredAssignments@408] - Checking task: task-003 
2016-12-18 19:58:21,586 - INFO  - [Thread-2-EventThread:RecoveredAssignments@408] - Checking task: task-002 
2016-12-18 19:58:21,586 - INFO  - [Thread-2-EventThread:RecoveredAssignments@412] - Size of tasks after status filtering: 0
2016-12-18 19:58:21,587 - INFO  - [Thread-2-EventThread:TestAssignmentRecovery$4@168] - Completed recovery: 0
2016-12-18 19:58:21,588 - INFO  - [Thread-2-EventThread:RecoveredAssignments$7@359] - Task correctly deleted: /tasks/task-001
2016-12-18 19:58:21,597 - INFO  - [Thread-2-EventThread:RecoveredAssignments$7@359] - Task correctly deleted: /tasks/task-003
2016-12-18 19:58:21,598 - INFO  - [Thread-2-EventThread:RecoveredAssignments$7@359] - Task correctly deleted: /tasks/task-002
2016-12-18 19:58:21,666 - INFO  - [main:NIOServerCnxn@1008] - Closed socket connection for client /127.0.0.1:43568 which had sessionid 0x159134c8e0d0000
2016-12-18 19:58:21,666 - INFO  - [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:11222:NIOServerCnxnFactory@219] - NIOServerCnxn factory exited run method
2016-12-18 19:58:21,666 - INFO  - [Thread-2-SendThread(localhost:11222):ClientCnxn$SendThread@1158] - Unable to read additional data from server sessionid 0x159134c8e0d0000, likely server has closed socket, closing socket connection and attempting reconnect
2016-12-18 19:58:21,667 - INFO  - [Standalone server with clientPort:11222:ZooKeeperServer@469] - shutting down
2016-12-18 19:58:21,667 - INFO  - [Standalone server with clientPort:11222:SessionTrackerImpl@225] - Shutting down
2016-12-18 19:58:21,667 - INFO  - [Standalone server with clientPort:11222:PrepRequestProcessor@767] - Shutting down
2016-12-18 19:58:21,667 - INFO  - [Standalone server with clientPort:11222:SyncRequestProcessor@209] - Shutting down
2016-12-18 19:58:21,667 - INFO  - [ProcessThread(sid:0 cport:11222)::PrepRequestProcessor@143] - PrepRequestProcessor exited loop!
2016-12-18 19:58:21,668 - INFO  - [SyncThread:0:SyncRequestProcessor@187] - SyncRequestProcessor exited!
2016-12-18 19:58:21,668 - INFO  - [Standalone server with clientPort:11222:FinalRequestProcessor@415] - shutdown of request processor complete
2016-12-18 19:58:21,672 - INFO  - [main:BaseTestCase@51] - Starting ZooKeeper Standalone Server: 11223
2016-12-18 19:58:21,673 - INFO  - [main:BaseTestCase@54] - ZooKeeper server started
2016-12-18 19:58:21,673 - WARN  - [Standalone server with clientPort:11223:ZooKeeperServerMain@76] - Unable to register log4j JMX control
javax.management.InstanceAlreadyExistsException: log4j:hiearchy=default
	at com.sun.jmx.mbeanserver.Repository.addMBean(Repository.java:437)
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerWithRepository(DefaultMBeanServerInterceptor.java:1898)
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerDynamicMBean(DefaultMBeanServerInterceptor.java:966)
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerObject(DefaultMBeanServerInterceptor.java:900)
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerMBean(DefaultMBeanServerInterceptor.java:324)
	at com.sun.jmx.mbeanserver.JmxMBeanServer.registerMBean(JmxMBeanServer.java:522)
	at org.apache.zookeeper.jmx.ManagedUtil.registerLog4jMBeans(ManagedUtil.java:53)
	at org.apache.zookeeper.server.ZooKeeperServerMain.initializeAndRun(ZooKeeperServerMain.java:74)
	at org.apache.zookeeper.book.StandaloneServer$TestZKSMain.initializeAndRun(StandaloneServer.java:51)
	at org.apache.zookeeper.book.StandaloneServer.run(StandaloneServer.java:97)
2016-12-18 19:58:21,674 - INFO  - [Standalone server with clientPort:11223:QuorumPeerConfig@103] - Reading configuration from: /tmp/test3423797008916674387.junit.dir/zoo.cfg
2016-12-18 19:58:21,674 - INFO  - [Thread-3:ZooKeeper@438] - Initiating client connection, connectString=localhost:11223 sessionTimeout=10000 watcher=org.apache.zookeeper.book.TestAssignmentRecovery$1@7b763267
2016-12-18 19:58:21,675 - INFO  - [Standalone server with clientPort:11223:ZooKeeperServerMain@95] - Starting server
2016-12-18 19:58:21,676 - INFO  - [Standalone server with clientPort:11223:ZooKeeperServer@787] - tickTime set to 2000
2016-12-18 19:58:21,676 - INFO  - [Standalone server with clientPort:11223:ZooKeeperServer@796] - minSessionTimeout set to -1
2016-12-18 19:58:21,676 - INFO  - [Standalone server with clientPort:11223:ZooKeeperServer@805] - maxSessionTimeout set to -1
2016-12-18 19:58:21,676 - INFO  - [Thread-3-SendThread(localhost:11223):ClientCnxn$SendThread@1032] - Opening socket connection to server localhost/127.0.0.1:11223. Will not attempt to authenticate using SASL (unknown error)
2016-12-18 19:58:21,677 - INFO  - [Standalone server with clientPort:11223:NIOServerCnxnFactory@89] - binding to port 0.0.0.0/0.0.0.0:11223
2016-12-18 19:58:21,678 - WARN  - [Thread-3-SendThread(localhost:11223):ClientCnxn$SendThread@1162] - Session 0x0 for server null, unexpected error, closing socket connection and attempting reconnect
java.net.ConnectException: Connexion refusée
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:361)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1141)
2016-12-18 19:58:21,767 - INFO  - [Thread-2-EventThread:TestAssignmentRecovery$3@133] - Event: WatchedEvent state:Disconnected type:None path:null
2016-12-18 19:58:22,000 - INFO  - [SessionTracker:SessionTrackerImpl@162] - SessionTrackerImpl exited loop!
2016-12-18 19:58:22,000 - INFO  - [SessionTracker:SessionTrackerImpl@162] - SessionTrackerImpl exited loop!
2016-12-18 19:58:22,779 - INFO  - [Thread-3-SendThread(localhost:11223):ClientCnxn$SendThread@1032] - Opening socket connection to server localhost/127.0.0.1:11223. Will not attempt to authenticate using SASL (unknown error)
2016-12-18 19:58:22,780 - INFO  - [Thread-3-SendThread(localhost:11223):ClientCnxn$SendThread@876] - Socket connection established to localhost/127.0.0.1:11223, initiating session
2016-12-18 19:58:22,780 - INFO  - [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:11223:NIOServerCnxnFactory@192] - Accepted socket connection from /127.0.0.1:46872
2016-12-18 19:58:22,782 - INFO  - [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:11223:ZooKeeperServer@900] - Client attempting to establish new session at /127.0.0.1:46872
2016-12-18 19:58:22,782 - INFO  - [SyncThread:0:FileTxnLog@199] - Creating new log file: log.1
2016-12-18 19:58:22,824 - INFO  - [SyncThread:0:ZooKeeperServer@645] - Established session 0x159134c8f6d0000 with negotiated timeout 10000 for client /127.0.0.1:46872
2016-12-18 19:58:22,824 - INFO  - [Thread-3-SendThread(localhost:11223):ClientCnxn$SendThread@1299] - Session establishment complete on server localhost/127.0.0.1:11223, sessionid = 0x159134c8f6d0000, negotiated timeout = 10000
2016-12-18 19:58:22,824 - INFO  - [Thread-3-EventThread:TestAssignmentRecovery$1@56] - Event: WatchedEvent state:SyncConnected type:None path:null
2016-12-18 19:58:23,008 - INFO  - [Thread-3-EventThread:RecoveredAssignments$1@108] - Getting tasks for recovery
2016-12-18 19:58:23,010 - INFO  - [Thread-3-EventThread:RecoveredAssignments$3@155] - Getting worker assignments for recovery: 1
2016-12-18 19:58:23,012 - INFO  - [Thread-3-EventThread:RecoveredAssignments$4@238] - Getting statuses for recovery
2016-12-18 19:58:23,053 - INFO  - [Thread-3-EventThread:RecoveredAssignments$7@359] - Task correctly deleted: /assign/worker-002/task-002
2016-12-18 19:58:23,062 - INFO  - [Thread-3-EventThread:RecoveredAssignments$7@359] - Task correctly deleted: /assign/worker-002
2016-12-18 19:58:23,062 - INFO  - [Thread-3-EventThread:RecoveredAssignments$8@384] - Processing assignments for recovery
2016-12-18 19:58:23,062 - INFO  - [Thread-3-EventThread:RecoveredAssignments@397] - Size of tasks: 3
2016-12-18 19:58:23,063 - INFO  - [Thread-3-EventThread:RecoveredAssignments@400] - Assignment: task-001
2016-12-18 19:58:23,063 - INFO  - [Thread-3-EventThread:RecoveredAssignments@405] - Size of tasks after assignment filtering: 2
2016-12-18 19:58:23,063 - INFO  - [Thread-3-EventThread:RecoveredAssignments@408] - Checking task: task-001 
2016-12-18 19:58:23,063 - INFO  - [Thread-3-EventThread:RecoveredAssignments@412] - Size of tasks after status filtering: 2
2016-12-18 19:58:23,064 - INFO  - [Thread-3-EventThread:TestAssignmentRecovery$2@102] - Completed recovery: 0
2016-12-18 19:58:23,064 - INFO  - [ProcessThread(sid:0 cport:11223)::PrepRequestProcessor@651] - Got user-level KeeperException when processing sessionid:0x159134c8f6d0000 type:delete cxid:0x17 zxid:0x12 txntype:-1 reqpath:n/a Error Path:/tasks/task-001 Error:KeeperErrorCode = NoNode for /tasks/task-001
2016-12-18 19:58:23,070 - INFO  - [Thread-3-EventThread:RecoveredAssignments$7@359] - Task correctly deleted: /tasks/task-001
2016-12-18 19:58:23,078 - ERROR - [Thread-3-EventThread:RecoveredAssignments$7@362] - Failed to delete task dataorg.apache.zookeeper.KeeperException$NoNodeException: KeeperErrorCode = NoNode for /tasks/task-001
2016-12-18 19:58:23,107 - INFO  - [main:NIOServerCnxn@1008] - Closed socket connection for client /127.0.0.1:46872 which had sessionid 0x159134c8f6d0000
2016-12-18 19:58:23,108 - INFO  - [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:11223:NIOServerCnxnFactory@219] - NIOServerCnxn factory exited run method
2016-12-18 19:58:23,108 - INFO  - [Thread-3-SendThread(localhost:11223):ClientCnxn$SendThread@1158] - Unable to read additional data from server sessionid 0x159134c8f6d0000, likely server has closed socket, closing socket connection and attempting reconnect
2016-12-18 19:58:23,108 - INFO  - [Standalone server with clientPort:11223:ZooKeeperServer@469] - shutting down
2016-12-18 19:58:23,109 - INFO  - [Standalone server with clientPort:11223:SessionTrackerImpl@225] - Shutting down
2016-12-18 19:58:23,109 - INFO  - [Standalone server with clientPort:11223:PrepRequestProcessor@767] - Shutting down
2016-12-18 19:58:23,109 - INFO  - [Standalone server with clientPort:11223:SyncRequestProcessor@209] - Shutting down
2016-12-18 19:58:23,109 - INFO  - [ProcessThread(sid:0 cport:11223)::PrepRequestProcessor@143] - PrepRequestProcessor exited loop!
2016-12-18 19:58:23,109 - INFO  - [SyncThread:0:SyncRequestProcessor@187] - SyncRequestProcessor exited!
2016-12-18 19:58:23,110 - INFO  - [Standalone server with clientPort:11223:FinalRequestProcessor@415] - shutdown of request processor complete
2016-12-18 19:58:23,112 - INFO  - [main:BaseTestCase@51] - Starting ZooKeeper Standalone Server: 11224
2016-12-18 19:58:23,113 - INFO  - [main:BaseTestCase@54] - ZooKeeper server started
2016-12-18 19:58:23,113 - WARN  - [Standalone server with clientPort:11224:ZooKeeperServerMain@76] - Unable to register log4j JMX control
javax.management.InstanceAlreadyExistsException: log4j:hiearchy=default
	at com.sun.jmx.mbeanserver.Repository.addMBean(Repository.java:437)
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerWithRepository(DefaultMBeanServerInterceptor.java:1898)
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerDynamicMBean(DefaultMBeanServerInterceptor.java:966)
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerObject(DefaultMBeanServerInterceptor.java:900)
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerMBean(DefaultMBeanServerInterceptor.java:324)
	at com.sun.jmx.mbeanserver.JmxMBeanServer.registerMBean(JmxMBeanServer.java:522)
	at org.apache.zookeeper.jmx.ManagedUtil.registerLog4jMBeans(ManagedUtil.java:53)
	at org.apache.zookeeper.server.ZooKeeperServerMain.initializeAndRun(ZooKeeperServerMain.java:74)
	at org.apache.zookeeper.book.StandaloneServer$TestZKSMain.initializeAndRun(StandaloneServer.java:51)
	at org.apache.zookeeper.book.StandaloneServer.run(StandaloneServer.java:97)
2016-12-18 19:58:23,114 - INFO  - [Standalone server with clientPort:11224:QuorumPeerConfig@103] - Reading configuration from: /tmp/test5135603909916447711.junit.dir/zoo.cfg
2016-12-18 19:58:23,115 - INFO  - [Standalone server with clientPort:11224:ZooKeeperServerMain@95] - Starting server
2016-12-18 19:58:23,115 - INFO  - [Standalone server with clientPort:11224:ZooKeeperServer@787] - tickTime set to 2000
2016-12-18 19:58:23,115 - INFO  - [Standalone server with clientPort:11224:ZooKeeperServer@796] - minSessionTimeout set to -1
2016-12-18 19:58:23,115 - INFO  - [Standalone server with clientPort:11224:ZooKeeperServer@805] - maxSessionTimeout set to -1
2016-12-18 19:58:23,116 - INFO  - [Standalone server with clientPort:11224:NIOServerCnxnFactory@89] - binding to port 0.0.0.0/0.0.0.0:11224
2016-12-18 19:58:23,115 - INFO  - [Thread-4:ZooKeeper@438] - Initiating client connection, connectString=localhost:11224 sessionTimeout=10000 watcher=org.apache.zookeeper.book.TestAssignmentRecovery$7@479de220
2016-12-18 19:58:23,117 - INFO  - [Thread-4-SendThread(localhost:11224):ClientCnxn$SendThread@1032] - Opening socket connection to server localhost/127.0.0.1:11224. Will not attempt to authenticate using SASL (unknown error)
2016-12-18 19:58:23,118 - INFO  - [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:11224:NIOServerCnxnFactory@192] - Accepted socket connection from /127.0.0.1:44750
2016-12-18 19:58:23,118 - INFO  - [Thread-4-SendThread(localhost:11224):ClientCnxn$SendThread@876] - Socket connection established to localhost/127.0.0.1:11224, initiating session
2016-12-18 19:58:23,119 - INFO  - [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:11224:ZooKeeperServer@900] - Client attempting to establish new session at /127.0.0.1:44750
2016-12-18 19:58:23,119 - INFO  - [SyncThread:0:FileTxnLog@199] - Creating new log file: log.1
2016-12-18 19:58:23,157 - INFO  - [SyncThread:0:ZooKeeperServer@645] - Established session 0x159134c950d0000 with negotiated timeout 10000 for client /127.0.0.1:44750
2016-12-18 19:58:23,157 - INFO  - [Thread-4-SendThread(localhost:11224):ClientCnxn$SendThread@1299] - Session establishment complete on server localhost/127.0.0.1:11224, sessionid = 0x159134c950d0000, negotiated timeout = 10000
2016-12-18 19:58:23,158 - INFO  - [Thread-4-EventThread:TestAssignmentRecovery$7@259] - Event: WatchedEvent state:SyncConnected type:None path:null
2016-12-18 19:58:23,209 - INFO  - [Thread-3-EventThread:TestAssignmentRecovery$1@56] - Event: WatchedEvent state:Disconnected type:None path:null
2016-12-18 19:58:23,363 - INFO  - [Thread-0-SendThread(localhost:11221):ClientCnxn$SendThread@1032] - Opening socket connection to server localhost/127.0.0.1:11221. Will not attempt to authenticate using SASL (unknown error)
2016-12-18 19:58:23,364 - WARN  - [Thread-0-SendThread(localhost:11221):ClientCnxn$SendThread@1162] - Session 0x159134c88bd0000 for server null, unexpected error, closing socket connection and attempting reconnect
java.net.ConnectException: Connexion refusée
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:361)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1141)
2016-12-18 19:58:23,373 - INFO  - [Thread-4-EventThread:RecoveredAssignments$1@108] - Getting tasks for recovery
2016-12-18 19:58:23,375 - INFO  - [Thread-4-EventThread:RecoveredAssignments$3@155] - Getting worker assignments for recovery: 1
2016-12-18 19:58:23,378 - INFO  - [Thread-4-EventThread:RecoveredAssignments$4@238] - Getting statuses for recovery
2016-12-18 19:58:23,379 - INFO  - [ProcessThread(sid:0 cport:11224)::PrepRequestProcessor@651] - Got user-level KeeperException when processing sessionid:0x159134c950d0000 type:delete cxid:0x12 zxid:0xd txntype:-1 reqpath:n/a Error Path:/assign/worker-002 Error:KeeperErrorCode = Directory not empty for /assign/worker-002
2016-12-18 19:58:23,431 - ERROR - [Thread-4-EventThread:RecoveredAssignments$7@362] - Failed to delete task dataorg.apache.zookeeper.KeeperException$NotEmptyException: KeeperErrorCode = Directory not empty for /assign/worker-002
2016-12-18 19:58:23,432 - INFO  - [Thread-4-EventThread:RecoveredAssignments$8@384] - Processing assignments for recovery
2016-12-18 19:58:23,432 - INFO  - [Thread-4-EventThread:RecoveredAssignments@397] - Size of tasks: 3
2016-12-18 19:58:23,432 - INFO  - [Thread-4-EventThread:RecoveredAssignments@400] - Assignment: task-001
2016-12-18 19:58:23,432 - INFO  - [Thread-4-EventThread:RecoveredAssignments@405] - Size of tasks after assignment filtering: 2
2016-12-18 19:58:23,432 - INFO  - [Thread-4-EventThread:RecoveredAssignments@412] - Size of tasks after status filtering: 2
2016-12-18 19:58:23,433 - INFO  - [Thread-4-EventThread:TestAssignmentRecovery$8@289] - Completed recovery: 0
2016-12-18 19:58:23,473 - INFO  - [main:NIOServerCnxn@1008] - Closed socket connection for client /127.0.0.1:44750 which had sessionid 0x159134c950d0000
2016-12-18 19:58:23,473 - INFO  - [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:11224:NIOServerCnxnFactory@219] - NIOServerCnxn factory exited run method
2016-12-18 19:58:23,473 - INFO  - [Thread-4-SendThread(localhost:11224):ClientCnxn$SendThread@1158] - Unable to read additional data from server sessionid 0x159134c950d0000, likely server has closed socket, closing socket connection and attempting reconnect
2016-12-18 19:58:23,474 - INFO  - [Standalone server with clientPort:11224:ZooKeeperServer@469] - shutting down
2016-12-18 19:58:23,474 - INFO  - [Standalone server with clientPort:11224:SessionTrackerImpl@225] - Shutting down
2016-12-18 19:58:23,474 - INFO  - [Standalone server with clientPort:11224:PrepRequestProcessor@767] - Shutting down
2016-12-18 19:58:23,474 - INFO  - [Standalone server with clientPort:11224:SyncRequestProcessor@209] - Shutting down
2016-12-18 19:58:23,474 - INFO  - [ProcessThread(sid:0 cport:11224)::PrepRequestProcessor@143] - PrepRequestProcessor exited loop!
2016-12-18 19:58:23,480 - ERROR - [SyncThread:0:NIOServerCnxn@178] - Unexpected Exception: 
java.nio.channels.CancelledKeyException
	at sun.nio.ch.SelectionKeyImpl.ensureValid(SelectionKeyImpl.java:73)
	at sun.nio.ch.SelectionKeyImpl.interestOps(SelectionKeyImpl.java:77)
	at org.apache.zookeeper.server.NIOServerCnxn.sendBuffer(NIOServerCnxn.java:151)
	at org.apache.zookeeper.server.NIOServerCnxn.sendResponse(NIOServerCnxn.java:1082)
	at org.apache.zookeeper.server.FinalRequestProcessor.processRequest(FinalRequestProcessor.java:404)
	at org.apache.zookeeper.server.SyncRequestProcessor.flush(SyncRequestProcessor.java:200)
	at org.apache.zookeeper.server.SyncRequestProcessor.run(SyncRequestProcessor.java:131)
2016-12-18 19:58:23,481 - INFO  - [SyncThread:0:SyncRequestProcessor@187] - SyncRequestProcessor exited!
2016-12-18 19:58:23,489 - ERROR - [Standalone server with clientPort:11224:NIOServerCnxn@178] - Unexpected Exception: 
java.nio.channels.CancelledKeyException
	at sun.nio.ch.SelectionKeyImpl.ensureValid(SelectionKeyImpl.java:73)
	at sun.nio.ch.SelectionKeyImpl.interestOps(SelectionKeyImpl.java:77)
	at org.apache.zookeeper.server.NIOServerCnxn.sendBuffer(NIOServerCnxn.java:151)
	at org.apache.zookeeper.server.NIOServerCnxn.sendResponse(NIOServerCnxn.java:1082)
	at org.apache.zookeeper.server.FinalRequestProcessor.processRequest(FinalRequestProcessor.java:404)
	at org.apache.zookeeper.server.SyncRequestProcessor.flush(SyncRequestProcessor.java:200)
	at org.apache.zookeeper.server.SyncRequestProcessor.shutdown(SyncRequestProcessor.java:216)
	at org.apache.zookeeper.server.PrepRequestProcessor.shutdown(PrepRequestProcessor.java:770)
	at org.apache.zookeeper.server.ZooKeeperServer.shutdown(ZooKeeperServer.java:480)
	at org.apache.zookeeper.server.ZooKeeperServerMain.runFromConfig(ZooKeeperServerMain.java:116)
	at org.apache.zookeeper.server.ZooKeeperServerMain.initializeAndRun(ZooKeeperServerMain.java:86)
	at org.apache.zookeeper.book.StandaloneServer$TestZKSMain.initializeAndRun(StandaloneServer.java:51)
	at org.apache.zookeeper.book.StandaloneServer.run(StandaloneServer.java:97)
2016-12-18 19:58:23,490 - INFO  - [Standalone server with clientPort:11224:FinalRequestProcessor@415] - shutdown of request processor complete
